src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuB2BShoppingServiceImpl.java

```diff
@@ -183,15 +192,60 @@ public List createRequisitionsFromCxml(B2BShoppingCart message, Person user) thr
 
             req.fixItemReferences();
 
-            // save requisition to database
-            purapService.saveDocumentNoValidation(req);
+            if (checkRequisitionAccountsAreUnique(req)) {
+                // save requisition to database
+                purapService.saveDocumentNoValidation(req);
 
-            // add requisition to List
-            requisitions.add(req);
+                // add requisition to List
+                requisitions.add(req);
+            }
+        }
+
+        if (Collections.isEmpty(requisitions)) {
+            throw new B2BShoppingException("Invalid e-shop Request. No requisitions were created.");
         }
+
         return requisitions;
     }
 
+    public boolean checkRequisitionAccountsAreUnique(RequisitionDocument requisitionDocument) {
+
+        List<RequisitionItem> requisitionItems = requisitionDocument.getItems();
+        for (RequisitionItem requisitionItem : requisitionItems) {
+
+            List<PurApAccountingLine> requisitionAccounts = requisitionItem.getSourceAccountingLines();
+
+            Set existingAccounts = new HashSet();
+            for(PurApAccountingLine purApAccountingLine : requisitionAccounts) {
+                if (!existingAccounts.contains(purApAccountingLine.toString())) {
+                    existingAccounts.add(purApAccountingLine.toString());
+                }
+                else {
+                    GlobalVariables.getMessageMap().putError(PurapConstants.ITEM_TAB_ERROR_PROPERTY, PurapKeyConstants.ERROR_ITEM_ACCOUNTING_NOT_UNIQUE,
+                            requisitionItem.getItemIdentifierString());
+                    String errorMessage = getDuplicateAccountErrorMessage(purApAccountingLine, requisitionItem);
+                    LOG.error(errorMessage);
+                    cuB2BShoppingErrorEmailService.sendDuplicateRequisitionAccountErrorEmail(requisitionDocument, errorMessage);
+                    return false;
```

Comment:
We may have cases where more than one item have duplicate accounts used.  I think it would be a good idea to report on all cases of duplicate accounts, not just the first.